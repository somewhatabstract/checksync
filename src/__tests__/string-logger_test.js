// @flow
import StringLogger from "../string-logger.js";

describe("StringLogger", () => {
    it.each(["log", "info", "warn", "error"])(
        "should add %s call to log",
        testCase => {
            // Arrange
            const logger = new StringLogger();

            // Act
            (logger: any)[testCase]("MESSAGE");
            const result = logger.getLog();

            // Assert
            expect(result).toMatchSnapshot(testCase);
        },
    );

    describe("#group", () => {
        it("should add group line to log", () => {
            // Arrange
            const logger = new StringLogger();

            // Act
            logger.group("GROUP!");
            const result = logger.getLog();

            // Assert
            expect(result).toMatchInlineSnapshot(`"<group>GROUP!"`);
        });

        it("should indent subsequent calls by group level", () => {
            // Arrange
            const logger = new StringLogger();

            // Act
            logger.group();
            logger.group();
            logger.group();
            logger.log("I'M INDENTED 3 times!");
            const results = logger.getLog().split("\n");

            // Assert
            expect(results[results.length - 1]).toStartWith("      ");
        });
    });

    describe("#groupEnd", () => {
        it("should not add anything if no group started", () => {
            // Arrange
            const logger = new StringLogger();

            // Act
            logger.groupEnd();
            const result = logger.getLog();

            // Assert
            expect(result).toBeEmpty();
        });

        it("should add group end line to log", () => {
            // Arrange
            const logger = new StringLogger();

            // Act
            logger.group("GROUP!");
            logger.groupEnd();
            const result = logger.getLog();

            // Assert
            expect(result).toMatchInlineSnapshot(`
                "<group>GROUP!
                <end_group>"
            `);
        });
    });
});
