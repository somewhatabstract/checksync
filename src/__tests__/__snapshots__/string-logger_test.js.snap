// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StringLogger should add error call to log: error 1`] = `" ERROR  MESSAGE"`;

exports[`StringLogger should add info call to log: info 1`] = `" INFO  MESSAGE"`;

exports[`StringLogger should add log call to log: log 1`] = `"MESSAGE"`;

exports[`StringLogger should add warn call to log: warn 1`] = `" WARNING  MESSAGE"`;
